slides/economic_scripting.tex

```diff
@@ -0,0 +1,378 @@
+			
+				\uncover<2 ->{\draw plot (-2,-3.1) node {\checkmarkgreen};}
+						
+				% bottom left
+				\uncover<3->{
+				\filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!25, thick, draw=highlight] (-2.6,-5.6) rectangle ++(3.3cm,1.8cm);
+				\filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!25, thick,draw=highlight] (-2.6,-5.6) rectangle ++(0.9cm,0.7cm);
+				\draw[color=black] plot (-2.5,-4.45)   node[right] {Bob:};
+				\draw[color=black] plot (-1.35,-4.45)   node[right] {\small{Colored Coin}};
+				\draw[color=black] plot (-2.5,-5.3)   node[right] {\small{$Sig_B$}};}
+				
+				\uncover<4->{\draw plot (-2,-5.3) node {\checkmarkgreen};}
+				
+				% Outputs
+				\draw[color=black] plot (3,-1.65)   node[above] {\texttt{outputs:}};
+				
+				% top right
+				\filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!25, thick, draw=highlight] (1.3,-3.4) rectangle ++(3.3cm,1.8cm);
+				\filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!25, thick, 	draw=highlight] (1.3,-3.4) rectangle ++(0.9cm,0.7cm);
+				\draw[color=black] plot (1.4,-2.25)   node[right] {Alice:};
+				\draw[color=black] plot (2.5,-2.25)   node[right] {\small{Colored Coin}};
+				\draw[color=black] plot (1.4,-3.1)   node[right] {\small{$Sig_A$}};
+				
+				% bottom right
+				\filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!25, thick, draw=highlight] (1.3,-5.6) rectangle ++(3.3cm,1.8cm);
+				\filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!25, thick,draw=highlight] (1.3,-5.6) rectangle ++(0.9cm,0.7cm);
+				\draw[color=black] plot (1.4,-4.45)   node[right] {Bob:};
+				\draw[color=black] plot (3.3,-4.45)   node[right] {3 BTC};
+				\draw[color=black] plot (1.4,-5.3)   node[right] {\small{$Sig_B$}};
+		\end{tikzpicture}
+	\end{figure}
+	\begin{itemize}
+		\item<1 ->Alice creates "wrapper" transaction and signs first input  (SIGHASH\_ALL)
+		\item<3 ->Bob signs second input (SIGHASH\_ALL)
+		\item<4 ->Atomic connection between the two assets' change of ownership. No counterparty risk.
+	\end{itemize}
+\end{frame}
+
+%%%
+
+\begin{frame}{Time Locks}
+	\begin{itemize}
+		\item<1-> UTXOs can be locked for a specified time
+		\begin{itemize}
+			\item<1-> Absolute time locks use a specific point in time or block height (analogy: "at 12 o'clock")
+			\item<1-> Relative time locks specify a number of blocks after confirmation (analogy: "in five hours")
+		\end{itemize}
+		\item<2 -> Four alternatives for implementation
+	\end{itemize}
+	\vspace{0.25cm}
+	\uncover<2->{\begin{table}
+		\center
+		\resizebox{\textwidth}{!}{%
+			\begin{tabular}{lll} \hline \hline
+				& Transaction level     & Output level\\
+				\cline{2-3}
+				& Consensus             & Scripting Language\\ \hline
+				Absolute time lock  & \texttt{nLockTime}    & CHECKLOCKTIMEVERIFY (CLTV)\\ 
+				Relative time lock  & \texttt{nSequence}    & CHECKSEQUENCEVERIFY (CSV)\\\hline \hline
+		\end{tabular}}
+	\end{table}}
+\end{frame}
```

Comment:
A "so-what" might add to the slide here

```suggestion
$\Rightarrow$ A signed transaction with a timelock promises a payment in the future, unless the referenced UTXO is spent otherwise in the meantime.
\end{frame}
```